src/simple_model.jl

```diff
@@ -99,9 +99,8 @@ function grad!(nlp :: SimpleNLPModel, x :: Vector, g :: Vector)
   if length(x) == length(g)
     return ForwardDiff.gradient!(g, nlp.f, x)
   else
-    gx = zeros(length(x))
+    gx = sub(g, 1:length(x))
     ForwardDiff.gradient!(gx, nlp.f, x)
-    g[1:length(x)] = gx
```

Comment:
Is the `if ... else` still needed, or can we always do `ForwardDiff.gradient!(sub(g, 1:length(x)), nlp.f, x)`?